packages/flutter/test/widgets/heroes_test.dart

```diff
@@ -1754,7 +1754,7 @@ Future<void> main() async {
       routes: routes,
       initialRoute: '/two',
     ));
-    expect(find.text('two'), findsOneWidget);
+    expect(tester.takeException(), isNull);
```

Comment:
This LGTM.  This was the actual intent of the test at the time - I probably either mistook which route actually had which widget, or was just doing this to make sure we didn't fire an assert.

Would it be worth testing here `expect(find.text('two'), findsNothing)` and `expect(find.text('three'), findsOneWidget)`?

Also, should we be worried about this breaking internal tests that relied on this behavior?

Reply:
Added those expects.

I think we just have to update those tests then. They are relying on a bug. Doing so is easy, see some of the other tests in this PR that I had to update.